tests/libpod/certs.go

```diff
@@ -0,0 +1,154 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package libpod
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+)
+
+// GetCertsForPods returns the used certificates for all pods matching  the label selector
+func GetCertsForPods(labelSelector string, namespace string, port string) ([][]byte, error) {
+	cli := kubevirt.Client()
+	pods, err := cli.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: labelSelector})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(pods.Items).ToNot(BeEmpty())
+
+	var certs [][]byte
+
+	//nolint:gocritic
+	for _, pod := range pods.Items {
+		podCopy := pod
+		err := func() error {
+			certs = append(certs, getCert(&podCopy, port))
+			return nil
+		}()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return certs, nil
+}
+
+//nolint:gosec
```

Comment:
Explanation for the nolint? Can it be added to specific lines?

Reply:
```
tests/libpod/certs.go:75:38: G404: Use of weak random number generator (math/rand instead of crypto/rand) (gosec)
	randPort := strconv.Itoa(basePort + rand.Intn(portRange))
	                                    ^
tests/libpod/certs.go:79:23: G402: TLS InsecureSkipVerify set true. (gosec)
		InsecureSkipVerify: true,
```

imo we should not add it as a comment, it is not a production code
and whoever need to know can remove and run lint or just check this thread

Reply:
done